contrib/syncthing/template.py

```diff
@@ -0,0 +1,72 @@
+pkgname = "syncthing"
+pkgver = "1.27.1"
+pkgrel = 0
+build_style = "go"
+make_build_args = [
+    f"-ldflags=-X github.com/syncthing/syncthing/lib/build.Version=v{pkgver}",
+]
+make_check_args = [
+    "./lib/...",
+    "./cmd/stdiscosrv",
+    "./cmd/strelaysrv",
+    "./cmd/syncthing",
+]
+hostmakedepends = ["go"]
+go_build_tags = ["noupgrade"]
+pkgdesc = "Continuous file synchronization program"
+maintainer = "triallax <triallax@tutanota.com>"
+license = "MPL-2.0"
+url = "https://syncthing.net"
+source = f"https://github.com/syncthing/{pkgname}/archive/v{pkgver}.tar.gz"
+sha256 = "e91671a8291a6d83264f1d1ed731fb6041b25d7259400e5f71cb2d241a48e6a4"
+
+
+def do_build(self):
+    self.do("go", "generate", "github.com/syncthing/syncthing/lib/api/auto")
+    self.golang.build(wrksrc="cmd/stdiscosrv")
+    self.golang.build(wrksrc="cmd/strelaysrv")
+    self.golang.build(wrksrc="cmd/syncthing")
```

Comment:
you should be able to put these in `make_build_args = ["./cmd/` and do a pre_build for `go generate`